syft/pkg/cataloger/golang/licenses.go

```diff
@@ -73,7 +63,7 @@ func newGoLicenseResolver(catalogerName string, opts CatalogerConfig) goLicenseR
 		opts:                  opts,
 		localModCacheDir:      localModCacheDir,
 		localVendorDir:        localVendorDir,
-		licenseCache:          cache.GetResolverCachingErrors[[]goLicense]("golang", "v1"),
+		licenseCache:          cache.GetResolverCachingErrors[[]pkg.License]("golang", "v1"),
```

Comment:
```suggestion
		licenseCache:          cache.GetResolverCachingErrors[[]pkg.License]("golang", "v2"),
```

Since you've changed the types we need to change the version here so that cache with the wrong data shape is not used.